NoSQL/Task1.md

```diff
@@ -0,0 +1,402 @@
+
+```
+cassandra@cqlsh:killrvideo> SELECT * FROM videos where video_id = 6c4cffb9-0dc4-1d59-af24-c960b5fc3652;
+
+ video_id                             | added_date                      | description
+
+
+
+
+
+                                                                             | title
+                                       | user_id
+--------------------------------------+---------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------+--------------------------------------
+ 6c4cffb9-0dc4-1d59-af24-c960b5fc3652 | 2014-11-06 01:11:50.000000+0000 | Speaker: Luke Tillman, Language Evangelist at DataStaxnnKeyboard Cat, Nyan Cat, and of course the world famous Grumpy Cat--it seems like the Internet can’t get enough cat videos. If you were building an application to let users share and consume their fill of videos, how would you go about it? In this talk, we’ll take a look at the data model for KillrVideo, a sample video sharing application similar to YouTube where users can share videos, comment, rate them, and more. You’ll learn get a practical introduction to Cassandra data modeling, querying with CQL, how the application drives the data model, and how to shift your thinking from the relational world you probably have experience with. | Cassandra Day Denver 2014: A Cassandra Data Model for Serving up Cat Videos | fd3f7889-fc0c-43db-951c-7b77710898bc
+
+(1 rows)
+```
+
+Next we Will explore some Other 
+CQL commands that will come in handy, like TRUNCATE in a later exercise, we will show you how to add/remove (non-primary key) columns. 
+## 11. Let's remove the data from our table using TRUNCATE .truncate videos; 
+
+```
+cassandra@cqlsh:killrvideo> TRUNCATE TABLE videos;
+```
+## 12. Create a second table in the 'killrvideo' keyspace called videos_by_title_year with the structure shown in above table. 
+Be sure users can query this table on both 'title' and 'added_year' by combining them into the partition key. 
+
+```
+cassandra@cqlsh:killrvideo> CREATE TABLE IF NOT EXISTS 
+videos_by_title_year (
+    title text,
+    added_year int,
+    added_date timestamp,
+    description text,
+    user_id uuid,
+    video_id timeuuid,
```

Comment:
Is there any reason to use timeuuid data type instead of uuid here?

Reply:
TIMEUUID is particular case of UUID (Type 1) which you can sort by timestamp, while UUID is just random unique key.
For clustering columns and sorting data it seems to be better to  use TIMEUUID for better performance.